### PR TITLE
fix: ref cols requested to sort desc where sorted asc

### PR DESCRIPTION
**Pull request by connoratrug**
ref cols requested to sort desc where sorted asc  

sort desc sorted ref in desc direction

- extract inner loop from orderBy builder
- rename isValidColumn to reflect its function
- begin adding tests to avoid regression

### CODE REVIEW

**Comment by connoratrug** [comment]
prepare code for adding https://github.com/molgenis/molgenis-emx2/pull/2482

File: backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryBuilderHelpersTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    final Reference ref1 = Mockito.mock(Reference.class);
+    when(ref1.getName()).thenReturn("ref1");
+    when(col1.getName()).thenReturn("col1");
+    when(col1.getReferences()).thenReturn(Collections.singletonList(ref1));
+    when(table.getColumns()).thenReturn(Collections.singletonList(col1));
+    when(table.getTableName()).thenReturn("testTable");
+    when(ref1.getPrimitiveType()).thenReturn(ColumnType.REF);
+
+    table.add(col1);
+    final Column column = SqlQueryBuilderHelpers.getColumnByName(table, "ref1");
```
Comment (mswertz) [inline backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryBuilderHelpersTest.java:56]: I must confess that I don't see the added value of these mock tests. If you need so many mocks on our own code I think we are testing the wrong things. Usually I would mock code on libraries from outside, not our own.

File: backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.molgenis.emx2.Column;
+import org.molgenis.emx2.Row;
+
+class SqlQueryTest {
+  @Test
+  void orderByTest() {
+
+    SqlSchemaMetadata schemaMock = Mockito.mock(SqlSchemaMetadata.class);
```
Comment (mswertz) [inline backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryTest.java:35]: Why don't you use a real schemametadata instance? There is no use to mock this?

This takes much more code and is much harder to follow what you are actually testing.

**Review by mswertz — CHANGES_REQUESTED** [review]
Sorry, I cannot see the bug you actually fixed. Probably because you did to many refactors for me to easily spot the bug being addressed. If I would have made these changes then I am quite sure your review would ask me to remove unnecessary changes. So I am surprised you now did this yourself.

Also most of this new test code is hard to follow making it hard to understand what you are actually testing.

As a general note, I rather test actual functionality then mocks (https://www.google.com/search?q=mocks+are+bad). Usually, it makes sense to mock external libraries but not to mock classes we actually build ourselves. You now waste test code on only these units while they actually could test multiple of our own classes at once. This tests don't give me confidence that the software is actually functional (only that the mocks and the tests match up) and mocks will be difficult to maintain.

As minimal change please explain to me what the actual bugfix is that I should be looking at. But ideally I would like to you to consider to rewrite these tests without using mocks, which should dramatically shorten them and make them actually test code behavior instead of mocks. I don't see the point of these mocks and I would prefer you to use one of the example models, ideally the Pet store. You don't even need to generate the backend schema for these tests I suppose.

P.S. make sure your PR name follows conventional commits standard because during merge we will squash the PR so it will become the commit message. I have updated this one.

File: backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    SelectSeekStep1<Record, Object> resultQueryMock =
+        Mockito.mock(SelectSeekStep1.class, withSettings().extraInterfaces(SelectJoinStep.class));
+    when(whereMock.orderBy((OrderField<Object>) any())).thenReturn(resultQueryMock);
+
+    Result<Record> fetchResult = Mockito.mock(Result.class);
+    when(resultQueryMock.fetch()).thenReturn(fetchResult);
+
+    when(fetchResult.iterator()).thenReturn(new ArrayList<Record>().iterator());
+
+    String flied = "name";
```
Comment (mswertz) [inline backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryTest.java:69]: field

File: backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    SqlQuery sqlQuery = new SqlQuery(schemaMock, flied);
```
Comment (mswertz) [inline backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryTest.java:70]: Here you could simply have used the pet store schema that is default already available right? Would have only taken 3 lines. And this bit is already under test in different places.

**Comment by mswertz** [comment]
> * begin adding tests to avoid regression

Did specifically you check pre-existing test coverage? I think much of SqlQuery is actually under test coverage.
I suppose you wanted to improve test coverage of the sort specifically?

https://sonarcloud.io/code?id=molgenis_molgenis-emx2&selected=molgenis_molgenis-emx2%3Abackend%2Fmolgenis-emx2-sql%2Fsrc%2Fmain%2Fjava%2Forg%2Fmolgenis%2Femx2%2Fsql%2FSqlQuery.java

**Comment by connoratrug** [comment]
i agree the code is untetable/unmaintainable and could be rewritten

File: backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryBuilderHelpersTest.java (hunk 1 above)
Comment (connoratrug) [reply backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryBuilderHelpersTest.java:56]: the mocks expose the code being tightly coupled
